llvm/lib/Transforms/Scalar/DivRemPairs.cpp

```diff
@@ -381,16 +381,15 @@ static bool optimizeDivRem(Function &F, const TargetTransformInfo &TTI,
       //   %mul = mul %div, 1   // %mul = undef
       //   %rem = sub %x, %mul  // %rem = undef - undef = undef
       // If X is not frozen, %rem becomes undef after transformation.
-      // TODO: We need a undef-specific checking function in ValueTracking
-      if (!isGuaranteedNotToBeUndefOrPoison(X, nullptr, DivInst, &DT)) {
+      if (!isGuaranteedNotToBeUndef(X, nullptr, DivInst, &DT)) {
```

Comment:
Need some tests.

Reply:
Working on the tests - I'm not sure what the new workflow for pre-commit tests is, since they usually seem to mention PR numbers.

Will try to see how to alive2 this on Monday - I don't usually hang out this high up the compiler infrastructure.

Reply:
Tests in https://github.com/llvm/llvm-project/pull/92628 - which I'm not landing because it's Friday evening

Reply:
See https://llvm.org/docs/InstCombineContributorGuide.html.

I believe it applies to all middle-end patches  :)

Reply:
Thanks! I'll do those things (which'll probably lead to improved tests for DivRemPairs overall, from what I can tell)

Reply:
Proofs at https://alive2.llvm.org/ce/z/9kR5LX